manifests/prepare.pp

```diff
@@ -0,0 +1,62 @@
+# == Class agent_upgrade::prepare
+#
+# This class is called from agent_upgrade to prepare for the upgrade.
+#
+class agent_upgrade::prepare {
+
+  File {
+    source_permissions => use,
+  }
+
+  # Migrate old files; assumes user Puppet runs under won't change during upgrade
+  file { ['/etc/puppetlabs', '/etc/puppetlabs/puppet']:
+    ensure => directory,
+  }
+
+  file { '/etc/puppetlabs/puppet/ssl':
+    ensure  => directory,
+    source  => $::puppet_ssldir,
+    backup  => false,
+    recurse => true,
+  }
+
+  $puppetconf = '/etc/puppetlabs/puppet/puppet.conf'
+  file { $puppetconf:
+    ensure => file,
+    source => $::puppet_config,
+  }
+
+  # manage puppet.conf contents, using inifile module
+  ['master', 'agent', 'main'].each |$section| {
```

Comment:
We expect to require future parser to be enabled before running the upgrade, as otherwise the user risks major breakages when they drop 3.x manifests into Puppet 4